matrix-api.c

```diff
@@ -81,8 +81,14 @@ void matrix_api_bad_response(MatrixConnectionData *ma, gpointer user_data,
                 _("Error from home server"), http_response_code);
     }
 
+    PurpleConnectionError error_reason = PURPLE_CONNECTION_ERROR_OTHER_ERROR;
```

Comment:
I seem to recall that this library avoids C99-style mid-function variable declarations. could you put this at the top of the function?